backend/src/polling_station/structs.rs

```diff
@@ -316,33 +443,211 @@ impl Validate for VotesCounts {
     #[schema(value_type = u32)]
     pub too_few_ballots_handed_out_count: Count,
-    /// Number of more ballots handed out ("Aantal keren dat er een stembiljet teveel is uitgereikt")
+    /// Number of more ballots handed out ("Hoe vaak is er een stembiljet te veel uitgereikt?")
     #[schema(value_type = u32)]
     pub too_many_ballots_handed_out_count: Count,
-    /// Number of other explanations ("Aantal keren dat er een andere verklaring is voor het verschil")
+    /// Number of other explanations ("Hoe vaak is er een andere verklaring voor het verschil?")
     #[schema(value_type = u32)]
     pub other_explanation_count: Count,
-    /// Number of no explanations ("Aantal keren dat er geen verklaring is voor het verschil")
+    /// Number of no explanations ("Hoe vaak is er geen verklaring voor het verschil?")
     #[schema(value_type = u32)]
     pub no_explanation_count: Count,
 }
 
-// TODO: impl Validate for DifferencesCounts
+impl Validate for DifferencesCounts {
+    fn validate(
+        &self,
+        election: &Election,
+        validation_results: &mut ValidationResults,
+        field_name: String,
+    ) {
+        // validate all counts
+        self.more_ballots_count.validate(
+            election,
+            validation_results,
+            format!("{field_name}.more_ballots_count"),
+        );
+        self.fewer_ballots_count.validate(
+            election,
+            validation_results,
+            format!("{field_name}.fewer_ballots_count"),
+        );
+        self.unreturned_ballots_count.validate(
+            election,
+            validation_results,
+            format!("{field_name}.unreturned_ballots_count"),
+        );
+        self.too_few_ballots_handed_out_count.validate(
+            election,
+            validation_results,
+            format!("{field_name}.too_few_ballots_handed_out_count"),
+        );
+        self.too_many_ballots_handed_out_count.validate(
+            election,
+            validation_results,
+            format!("{field_name}.too_many_ballots_handed_out_count"),
+        );
+        self.other_explanation_count.validate(
+            election,
+            validation_results,
+            format!("{field_name}.other_explanation_count"),
+        );
+        self.no_explanation_count.validate(
+            election,
+            validation_results,
+            format!("{field_name}.no_explanation_count"),
+        );
+
+        // W.302 validate that more_ballots_count == too_many_ballots_handed_out_count + other_explanation_count + no_explanation_count - unreturned_ballots_count - too_few_ballots_handed_out_count
+        if self.more_ballots_count != 0
+            && (self.too_many_ballots_handed_out_count
+                + self.other_explanation_count
+                + self.no_explanation_count
+                - self.unreturned_ballots_count
+                - self.too_few_ballots_handed_out_count
+                != self.more_ballots_count)
```

Comment:
These are `u32`s, so if `unreturned_ballots_count` and `too_few_ballots_handed_out_count` are larger than the other fields, it results in:
```
thread 'tokio-runtime-worker' panicked at src/polling_station/structs.rs:609:17:
attempt to subtract with overflow
```

I expect the same to be the case for W.303.

Reply:
When adding up those counts you should be using `u64` like in some other places.

Reply:
I think that's a different (and equally) valid issue:
- because we're adding up `u32`s, we might get something bigger than a `u32`
- because we're subtracting `u32`s, we might get a negative number

Reply:
Oh right, for the subtraction `i64` is more suitable.

Reply:
Fixed and tested in 6bc986da52492c8a4ad8a5782d1ee1309855ce5e